### ISSUE TITLE
Improve logging of monitor validation failures

### ISSUE THREAD

**Issue by keeganwitt** [DataDog/terraform-provider-datadog#1916]
### Terraform Version

1.3.6

### Affected Resource(s)

-   datadog_monitor

### Expected Behavior

The missing or misconfigured portion of the monitor is mentioned in some way.

### Actual Behavior

It just prints the error from the API endpoint and I have no idea where the problem is with my monitor.

```
Error: error validating monitor from https://api.datadoghq.com/api/v1/monitor/validate: 400 Bad Request: {"errors":["Alert status is required."]}
```

At the very least, printing the monitor struct where this error is thrown (https://github.com/DataDog/terraform-provider-datadog/blob/06f894711e87f8bd7dd094d38138711fae5aa173/datadog/resource_datadog_monitor.go#L829) would provide at least some context.

**Comment by nkzou** [comment]
Unfortunately on the terraform provider side we don't have any more specific debug info - all we know on our end is what the monitor validation endpoint returns to us. You can try using the environment var `TF_LOG=DEBUG` to see the exact http request that's being made, but otherwise we have no more info to log.

You can contact Datadog support to make a feature request for better error messages from the monitor validation endpoint, or to get help with debugging the monitor definition itself. You could also try creating the monitor in the web app first, exporting it as JSON, and using that output to craft the definition.

As for this specific error (`"Alert status is required"`), I managed to reproduce it - it occurs when `critical_recovery` is set, but `critical` isn't. I don't know for sure if this is your issue, but hopefully that helps.

**Comment by nkzou** [comment]
Just to confirm, when you do `TF_LOG=DEBUG terraform apply` you don't see anything similar to the following? -
```
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: 2023/05/12 14:57:17
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: POST /api/v1/monitor/118931413/validate HTTP/1.1
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Host: api.datadoghq.com
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: User-Agent: terraform-provider-datadog/3.25.0 (terraform 2.10.1; terraform-cli 1.3.7) datadog-api-client-go/2.12.1+dev (go go1.19.8; os darwin; arch arm64)
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Content-Length: 368
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Accept: application/json
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Content-Type: application/json
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Dd-Api-Key: REDACTED
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Dd-Application-Key: REDACTED
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: Accept-Encoding: gzip
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0:
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0: {"message":"","name":"evalwindownulltest","options":{"include_tags":true,"new_host_delay":300,"no_data_timeframe":10,"notify_no_data":false,"require_full_window":true,"thresholds":{"critical_recovery":4,"warning":3,"warning_recovery":2}},"priority":0,"query":"avg(last_1h):avg:aws.ec2.cpu{environment:foo,host:foo} by {host} \u003e 4","tags":[],"type":"metric alert"}
2023-05-12T14:57:17.719-0400 [DEBUG] provider.terraform-provider-datadog_v3.25.0:
```
Do you see terraform debug output at all, or is it just these request/response logs that seem to be missing?

**Comment by keeganwitt** [comment]
That's right, but I was running `plan`, not `apply`. Not sure if that makes a difference.

**Comment by keeganwitt** [comment]
I'm suspecting when one of the thresholds gets set to null the request gets set to something invalid. If I'm right about that, that'd be a separate bug.

**Comment by keeganwitt** [comment]
Just to conclude the sidebar of what caused the issue prompting this: the loop creates a monitor with only a warning level set and no critical level, which is apparently not allowed.

I still do have no explanation for why `plan` doesn't print the JSON requests to the Datadog API.

**Comment by davidjrice** [comment]
👍 I think improving the logging would be great, I only managed to realise that the message "Alert status is required" was misleading and address my actual issue when trying to create a new monitor thanks to the hard work by @keeganwitt which ended up being that creating an `error-tracking alert` monitor it is required to set the `critical` monitor threshold to `null`

**Comment by nkzou** [comment]
@davidjrice Does running the plan with the env var `TF_LOG=DEBUG` properly log http request/responses for you? If it doesn't then there seems to be something going wrong with certain environments/setups.

The underlying problem here is that the monitor validation endpoint doesn't return enough debugging output. I've let the team know. For now, on the terraform provider side, we don't have any more info available to log about the error outside of the endpoint response (the "alert status is required" message).
